### PR TITLE
Add fallback to CBS extractor

### PR DESCRIPTION
**Pull request by llacb47**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Possibly fixes https://github.com/ytdl-org/youtube-dl/issues/29564 (it does for the video OP tried to download)

The CBS/paramountplus extractor currently relies solely on the `videoPlayerService` endpoint, however for some videos that endpoint doesn't provide the data we need. 

```
>curl "https://can.cbs.com/thunder/player/videoPlayerService.php?partner=cbs&contentId=Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k"
<?xml version="1.0" encoding="UTF-8" standalone="yes"?><data><items><item><showPageURL>http://www.cbs.com</showPageURL><videoLength>0</videoLength><isServiceAllowed>false</isServiceAllowed><code>315</code><contentId>Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k</contentId><tvRatingFlag>true</tvRatingFlag><bitrate>0</bitrate><excludeOztam>false</excludeOztam></item></items></data>
```

This patch adds a fallback to `_extract_theplatform_smil` and allows extraction for these videos to succeed. 
Verbose logs with this patch:
```
./yt-dlp.cmd --ignore-config --proxy "http://localhost:8888/" -v "https://www.paramountplus.com/shows/catdog/video/Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k/catdog-climb-every-catdog-the-canine-mutiny/" -F
[debug] Command-line config: ['--ignore-config', '--proxy', 'http://localhost:8888/', '-v', 'https://www.paramountplus.com/shows/catdog/video/Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k/catdog-climb-every-catdog-the-canine-mutiny/', '-F']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] yt-dlp version 2021.07.24 (source)
[debug] Plugin Extractors: ['SamplePlugin']
[debug] Git HEAD: cf2226e24
[debug] Python version 3.9.6 (CPython 64bit) - Windows-10-10.0.19043-SP0
[debug] exe versions: ffmpeg n4.3.2-163-g6c414cf8f7, ffprobe n4.3.2-163-g6c414cf8f7
[debug] Proxy map: {'http': 'http://localhost:8888/', 'https': 'http://localhost:8888/'}
[debug] [CBS] Extracting URL: https://www.paramountplus.com/shows/catdog/video/Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k/catdog-climb-every-catdog-the-canine-mutiny/
[CBS] Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k: Downloading XML
[CBS] Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k: Downloading fallback SMIL data
[CBS] Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[CBS] Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k: Downloading JSON metadata
[info] Available formats for Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k:
ID       EXT RESOLUTION |   TBR PROTO  | VCODEC        VBR ACODEC     ABR
-------- --- ---------- - ----- ------ - ----------- ----- --------- ----
hls-93   mp4 audio only |   93k m3u8_n |                   mp4a.40.2  93k
hls-496  mp4 640x480    |  496k m3u8_n | avc1.640028  496k mp4a.40.2   0k
hls-4612 mp4 720x540    | 4612k m3u8_n | avc1.640028 4612k mp4a.40.2   0k
hls-841  mp4 768x576    |  841k m3u8_n | avc1.640028  841k mp4a.40.2   0k
hls-1618 mp4 960x720    | 1618k m3u8_n | avc1.640028 1618k mp4a.40.2   0k
hls-2216 mp4 1024x768   | 2216k m3u8_n | avc1.640028 2216k mp4a.40.2   0k
hls-3113 mp4 1280x960   | 3113k m3u8_n | avc1.640028 3113k mp4a.40.2   0k
```

```
./yt-dlp.cmd --ignore-config --proxy "http://localhost:8888/" -v -F "https://www.paramountplus.com/shows/tooning-out-the-news/video/6hSWYWRrR9EUTz7IEe5fJKBhYvSUfexd/7-23-21-week-in-review-rep-jahana-hayes-howard-fineman-sen-michael-bennet-sheera-frenkel-cecilia-kang-/"
[debug] Command-line config: ['--ignore-config', '--proxy', 'http://localhost:8888/', '-v', '-F', 'https://www.paramountplus.com/shows/tooning-out-the-news/video/6hSWYWRrR9EUTz7IEe5fJKBhYvSUfexd/7-23-21-week-in-review-rep-jahana-hayes-howard-fineman-sen-michael-bennet-sheera-frenkel-cecilia-kang-/']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] yt-dlp version 2021.07.24 (source)
[debug] Plugin Extractors: ['SamplePlugin']
[debug] Git HEAD: cf2226e24
[debug] Python version 3.9.6 (CPython 64bit) - Windows-10-10.0.19043-SP0
[debug] exe versions: ffmpeg n4.3.2-163-g6c414cf8f7, ffprobe n4.3.2-163-g6c414cf8f7
[debug] Proxy map: {'http': 'http://localhost:8888/', 'https': 'http://localhost:8888/'}
[debug] [CBS] Extracting URL: https://www.paramountplus.com/shows/tooning-out-the-news/video/6hSWYWRrR9EUTz7IEe5fJKBhYvSUfexd/7-23-21-week-in-review-rep-jahana-hayes-howard-fineman-sen-michael-bennet-sheera-frenkel-cecilia-kang-/
[CBS] 6hSWYWRrR9EUTz7IEe5fJKBhYvSUfexd: Downloading XML
[CBS] 6hSWYWRrR9EUTz7IEe5fJKBhYvSUfexd: Downloading fallback SMIL data
[CBS] 6hSWYWRrR9EUTz7IEe5fJKBhYvSUfexd: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[CBS] 6hSWYWRrR9EUTz7IEe5fJKBhYvSUfexd: Downloading JSON metadata
[info] Available formats for 6hSWYWRrR9EUTz7IEe5fJKBhYvSUfexd:
ID       EXT RESOLUTION |   TBR PROTO  | VCODEC        VBR ACODEC     ABR
-------- --- ---------- - ----- ------ - ----------- ----- --------- ----
hls-93   mp4 audio only |   93k m3u8_n |                   mp4a.40.2  93k
hls-485  mp4 640x360    |  485k m3u8_n | avc1.640028  485k mp4a.40.2   0k
hls-823  mp4 768x432    |  823k m3u8_n | avc1.640028  823k mp4a.40.2   0k
hls-1586 mp4 960x540    | 1586k m3u8_n | avc1.640028 1586k mp4a.40.2   0k
hls-2175 mp4 1024x576   | 2175k m3u8_n | avc1.640028 2175k mp4a.40.2   0k
hls-3041 mp4 1280x720   | 3041k m3u8_n | avc1.640028 3041k mp4a.40.2   0k
hls-4507 mp4 1920x1080  | 4507k m3u8_n | avc1.640028 4507k mp4a.40.2   0k
```
```
./yt-dlp.cmd --ignore-config --proxy "http://localhost:8888/" -v -F "https://www.cbs.com/shows/the-late-show-with-stephen-colbert/video/60icOhMb9NcjbcWnF_gub9XXHdeBcNk2/the-late-show-6-23-21-christine-baranski-joy-oladokun-"
[debug] Command-line config: ['--ignore-config', '--proxy', 'http://localhost:8888/', '-v', '-F', 'https://www.cbs.com/shows/the-late-show-with-stephen-colbert/video/60icOhMb9NcjbcWnF_gub9XXHdeBcNk2/the-late-show-6-23-21-christine-baranski-joy-oladokun-']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] yt-dlp version 2021.07.24 (source)
[debug] Plugin Extractors: ['SamplePlugin']
[debug] Git HEAD: cf2226e24
[debug] Python version 3.9.6 (CPython 64bit) - Windows-10-10.0.19043-SP0
[debug] exe versions: ffmpeg n4.3.2-163-g6c414cf8f7, ffprobe n4.3.2-163-g6c414cf8f7
[debug] Proxy map: {'http': 'http://localhost:8888/', 'https': 'http://localhost:8888/'}
[debug] [CBS] Extracting URL: https://www.cbs.com/shows/the-late-show-with-stephen-colbert/video/60icOhMb9NcjbcWnF_gub9XXHdeBcNk2/the-late-show-6-23-21-christine-baranski-joy-oladokun-
[CBS] 60icOhMb9NcjbcWnF_gub9XXHdeBcNk2: Downloading XML
[CBS] 60icOhMb9NcjbcWnF_gub9XXHdeBcNk2: Downloading fallback SMIL data
[CBS] 60icOhMb9NcjbcWnF_gub9XXHdeBcNk2: Downloading fallback SMIL data, trying again with another format
ERROR: This content expired on 7/24/21 7:35 AM.
Traceback (most recent call last):
  File "E:\Documents\git\yt-dlp\yt_dlp\extractor\cbs.py", line 109, in _extract_video_info
    tp_formats, tp_subtitles = self._extract_theplatform_smil(
  File "E:\Documents\git\yt-dlp\yt_dlp\extractor\theplatform.py", line 51, in _extract_theplatform_smil
    raise ExtractorError(
yt_dlp.utils.ExtractorError: None of the available releases match the specified AssetType, ProtectionScheme, and/or Format preferences
Traceback (most recent call last):
  File "E:\Documents\git\yt-dlp\yt_dlp\extractor\cbs.py", line 109, in _extract_video_info
    tp_formats, tp_subtitles = self._extract_theplatform_smil(
  File "E:\Documents\git\yt-dlp\yt_dlp\extractor\theplatform.py", line 51, in _extract_theplatform_smil
    raise ExtractorError(
yt_dlp.utils.ExtractorError: None of the available releases match the specified AssetType, ProtectionScheme, and/or Format preferences

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "E:\Documents\git\yt-dlp\yt_dlp\YoutubeDL.py", line 1152, in wrapper
    return func(self, *args, **kwargs)
  File "E:\Documents\git\yt-dlp\yt_dlp\YoutubeDL.py", line 1177, in __extract_info
    ie_result = ie.extract(url)
  File "E:\Documents\git\yt-dlp\yt_dlp\extractor\common.py", line 572, in extract
    ie_result = self._real_extract(url)
  File "E:\Documents\git\yt-dlp\yt_dlp\extractor\cbs.py", line 145, in _real_extract
    return self._extract_video_info(content_id)
  File "E:\Documents\git\yt-dlp\yt_dlp\extractor\cbs.py", line 115, in _extract_video_info
    tp_formats, tp_subtitles = self._extract_theplatform_smil(
  File "E:\Documents\git\yt-dlp\yt_dlp\extractor\theplatform.py", line 51, in _extract_theplatform_smil
    raise ExtractorError(
yt_dlp.utils.ExtractorError: This content expired on 7/24/21 7:35 AM.
```

### CODE REVIEW

File: yt_dlp/extractor/cbs.py [hunk 1]
```diff
@@ -78,16 +78,29 @@ def _extract_video_info(self, content_id, site='cbs', mpx_acc=2198311517):
...
             query = {
                 'mbr': 'true',
                 'assetTypes': asset_type,
             }
+            if not asset_type:
+                # fallback for content_ids that videoPlayerService doesn't return anything for
+                # Examples:
+                # https://www.paramountplus.com/shows/catdog/video/Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k/catdog-climb-every-catdog-the-canine-mutiny/
+                # https://www.paramountplus.com/shows/tooning-out-the-news/video/6hSWYWRrR9EUTz7IEe5fJKBhYvSUfexd/7-23-21-week-in-review-rep-jahana-hayes-howard-fineman-sen-michael-bennet-sheera-frenkel-cecilia-kang-/
+                # https://www.cbs.com/shows/the-late-show-with-stephen-colbert/video/60icOhMb9NcjbcWnF_gub9XXHdeBcNk2/the-late-show-6-23-21-christine-baranski-joy-oladokun- (Expired)
```
Comment (pukkandan) [inline yt_dlp/extractor/cbs.py:94]: remove comments and add these as tests

File: yt_dlp/extractor/cbs.py [hunk 2]
```diff
@@ -78,16 +78,29 @@ def _extract_video_info(self, content_id, site='cbs', mpx_acc=2198311517):
...
+                useXMLmetadata = False
+                asset_type = 'fallback'
+                formatList = ['M3U+none,MPEG4,M3U+appleHlsEncryption,MP3', '']  # blank query to check if expired
+                currentIndex = 0
+                query['formats'] = formatList[currentIndex]
+                del query['assetTypes']
+            if asset_type in asset_types or 'HLS_FPS' in asset_type or 'DASH_CENC' in asset_type or '':
```
Comment (pukkandan) [inline yt_dlp/extractor/cbs.py:101]:
```suggestion
            elif asset_type in asset_types or 'HLS_FPS' in asset_type or 'DASH_CENC' in asset_type:
```

File: yt_dlp/extractor/cbs.py [hunk 3]
```diff
@@ -97,6 +110,11 @@ def _extract_video_info(self, content_id, site='cbs', mpx_acc=2198311517):
                     update_url_query(tp_release_url, query), content_id,
                     'Downloading %s SMIL data' % asset_type)
             except ExtractorError as e:
+                if not useXMLmetadata and currentIndex + 1 is not len(formatList):
```
Comment (pukkandan) [inline yt_dlp/extractor/cbs.py:113]:
```suggestion
                if not useXMLmetadata and currentIndex + 1 < len(formatList):
```

File: yt_dlp/extractor/cbs.py [hunk 4]
```diff
@@ -78,16 +78,29 @@ def _extract_video_info(self, content_id, site='cbs', mpx_acc=2198311517):
...
+            if not asset_type:
+                # fallback for content_ids that videoPlayerService doesn't return anything for
+                # Examples:
+                # https://www.paramountplus.com/shows/catdog/video/Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k/catdog-climb-every-catdog-the-canine-mutiny/
+                # https://www.paramountplus.com/shows/tooning-out-the-news/video/6hSWYWRrR9EUTz7IEe5fJKBhYvSUfexd/7-23-21-week-in-review-rep-jahana-hayes-howard-fineman-sen-michael-bennet-sheera-frenkel-cecilia-kang-/
+                # https://www.cbs.com/shows/the-late-show-with-stephen-colbert/video/60icOhMb9NcjbcWnF_gub9XXHdeBcNk2/the-late-show-6-23-21-christine-baranski-joy-oladokun- (Expired)
+                useXMLmetadata = False
+                asset_type = 'fallback'
+                formatList = ['M3U+none,MPEG4,M3U+appleHlsEncryption,MP3', '']  # blank query to check if expired
+                currentIndex = 0
```
Comment (pukkandan) [inline yt_dlp/extractor/cbs.py:98]: currentIndex is never changed?

**Comment by llacb47** [comment]
`OnceURL` is defunct now (`once.unicornmedia.com` is dead) so I added it to the "formats to ignore" list.

**Comment by llacb47** [comment]
I don't think the last test I added is working right. Is there a trick to `expected_warnings`?

**Comment by llacb47** [comment]
all of this should be removed as well https://github.com/yt-dlp/yt-dlp/search?q=OnceIE

**Comment by pukkandan** [comment]
`expected_warnings` only catches warnings, not errors

File: yt_dlp/extractor/cbs.py [hunk 5]
```diff
@@ -97,6 +146,11 @@ def _extract_video_info(self, content_id, site='cbs', mpx_acc=2198311517):
                     update_url_query(tp_release_url, query), content_id,
                     'Downloading %s SMIL data' % asset_type)
             except ExtractorError as e:
+                if not useXMLmetadata:
+                    query['formats'] = formatList[1]
+                    tp_formats, tp_subtitles = self._extract_theplatform_smil(
+                        update_url_query(tp_release_url, query), content_id,
+                        'Downloading %s SMIL data, trying again with another format' % asset_type)
                 last_e = e
                 continue
```
Comment (pukkandan) [inline yt_dlp/extractor/cbs.py:155]: This should be removed, no?

Comment (llacb47) [reply yt_dlp/extractor/cbs.py]: Yeah, it really serves no purpose as the video can't be downloaded anyway if formats = `'M3U+none,MPEG4,M3U+appleHlsEncryption,MP3'` doesn't yield anything from theplatform's API. Only used to tell if the video expired, but it's not really important. Do you want me to remove the entire formatList / `if not useXMLmetadata:` section to simplify things?

**Comment by pukkandan** [comment]
Seems I may have misunderstood the purpose of the PR?

The tests you added works even without this

```cmd
$ yt-dlp https://www.paramountplus.com/shows/tooning-out-the-news/video/6hSWYWRrR9EUTz7IEe5fJKBhYvSUfexd/7-23-21-week-in-review-rep-jahana-hayes-howard-fineman-sen-michael-bennet-sheera-frenkel-cecilia-kang- https://www.paramountplus.com/shows/catdog/video/Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k/catdog-climb-every-catdog-the-canine-mutiny https://www.cbs.com/shows/the-late-show-with-stephen-colbert/video/60icOhMb9NcjbcWnF_gub9XXHdeBcNk2/the-late-show-6-23-21-christine-baranski-joy-oladokun- -vF
[debug] Command-line config: ['https://www.paramountplus.com/shows/tooning-out-the-news/video/6hSWYWRrR9EUTz7IEe5fJKBhYvSUfexd/7-23-21-week-in-review-rep-jahana-hayes-howard-fineman-sen-michael-bennet-sheera-frenkel-cecilia-kang-', 'https://www.paramountplus.com/shows/catdog/video/Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k/catdog-climb-every-catdog-the-canine-mutiny', 'https://www.cbs.com/shows/the-late-show-with-stephen-colbert/video/60icOhMb9NcjbcWnF_gub9XXHdeBcNk2/the-late-show-6-23-21-christine-baranski-joy-oladokun-', '-vF']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] yt-dlp version 2021.07.24 (source)
[debug] Plugin Extractors: ['SamplePlugin']
[debug] Git HEAD: 2a9c6dcd22
[debug] Python version 3.9.6 (CPython 64bit) - Windows-10-10.0.19043-SP0
[debug] exe versions: ffmpeg 2021-04-11-git-309e3cc15c-full_build-www.gyan.dev, ffprobe 2021-04-11-git-309e3cc15c-full_build-www.gyan.dev, phantomjs 2.1.1
[debug] Proxy map: {}
[debug] [CBS] Extracting URL: https://www.paramountplus.com/shows/tooning-out-the-news/video/6hSWYWRrR9EUTz7IEe5fJKBhYvSUfexd/7-23-21-week-in-review-rep-jahana-hayes-howard-fineman-sen-michael-bennet-sheera-frenkel-cecilia-kang-
[CBS] 6hSWYWRrR9EUTz7IEe5fJKBhYvSUfexd: Downloading XML
[CBS] 6hSWYWRrR9EUTz7IEe5fJKBhYvSUfexd: Downloading StreamPack SMIL data
[CBS] 6hSWYWRrR9EUTz7IEe5fJKBhYvSUfexd: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[CBS] 6hSWYWRrR9EUTz7IEe5fJKBhYvSUfexd: Downloading JSON metadata
[info] Available formats for 6hSWYWRrR9EUTz7IEe5fJKBhYvSUfexd:
ID       EXT RESOLUTION |   TBR PROTO  | VCODEC        VBR ACODEC     ABR
-------- --- ---------- - ----- ------ - ----------- ----- --------- ----
hls-93   mp4 audio only |   93k m3u8_n |                   mp4a.40.2  93k
hls-485  mp4 640x360    |  485k m3u8_n | avc1.640028  485k mp4a.40.2   0k
hls-823  mp4 768x432    |  823k m3u8_n | avc1.640028  823k mp4a.40.2   0k
hls-1586 mp4 960x540    | 1586k m3u8_n | avc1.640028 1586k mp4a.40.2   0k
hls-2175 mp4 1024x576   | 2175k m3u8_n | avc1.640028 2175k mp4a.40.2   0k
hls-3041 mp4 1280x720   | 3041k m3u8_n | avc1.640028 3041k mp4a.40.2   0k
hls-4507 mp4 1920x1080  | 4507k m3u8_n | avc1.640028 4507k mp4a.40.2   0k
[debug] [CBS] Extracting URL: https://www.paramountplus.com/shows/catdog/video/Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k/catdog-climb-every-catdog-the-canine-mutiny
[CBS] Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k: Downloading XML
[CBS] Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k: Downloading StreamPack SMIL data
[CBS] Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[CBS] Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k: Downloading JSON metadata
[info] Available formats for Oe44g5_NrlgiZE3aQVONleD6vXc8kP0k:
ID       EXT RESOLUTION |   TBR PROTO  | VCODEC        VBR ACODEC     ABR
-------- --- ---------- - ----- ------ - ----------- ----- --------- ----
hls-93   mp4 audio only |   93k m3u8_n |                   mp4a.40.2  93k
hls-496  mp4 640x480    |  496k m3u8_n | avc1.640028  496k mp4a.40.2   0k
hls-4612 mp4 720x540    | 4612k m3u8_n | avc1.640028 4612k mp4a.40.2   0k
hls-841  mp4 768x576    |  841k m3u8_n | avc1.640028  841k mp4a.40.2   0k
hls-1618 mp4 960x720    | 1618k m3u8_n | avc1.640028 1618k mp4a.40.2   0k
hls-2216 mp4 1024x768   | 2216k m3u8_n | avc1.640028 2216k mp4a.40.2   0k
hls-3113 mp4 1280x960   | 3113k m3u8_n | avc1.640028 3113k mp4a.40.2   0k
[debug] [CBS] Extracting URL: https://www.cbs.com/shows/the-late-show-with-stephen-colbert/video/60icOhMb9NcjbcWnF_gub9XXHdeBcNk2/the-late-show-6-23-21-christine-baranski-joy-oladokun-
[CBS] 60icOhMb9NcjbcWnF_gub9XXHdeBcNk2: Downloading XML
ERROR: No video formats found; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "D:\Programs\Source\yt-dlp\yt_dlp\YoutubeDL.py", line 1152, in wrapper
    return func(self, *args, **kwargs)
  File "D:\Programs\Source\yt-dlp\yt_dlp\YoutubeDL.py", line 1177, in __extract_info
    ie_result = ie.extract(url)
  File "D:\Programs\Source\yt-dlp\yt_dlp\extractor\common.py", line 572, in extract
    ie_result = self._real_extract(url)
  File "D:\Programs\Source\yt-dlp\yt_dlp\extractor\cbs.py", line 124, in _real_extract
    return self._extract_video_info(content_id)
  File "D:\Programs\Source\yt-dlp\yt_dlp\extractor\cbs.py", line 106, in _extract_video_info
    self._sort_formats(formats)
  File "D:\Programs\Source\yt-dlp\yt_dlp\extractor\common.py", line 1750, in _sort_formats
    raise ExtractorError('No video formats found')
yt_dlp.utils.ExtractorError: No video formats found; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
```

**Comment by llacb47** [comment]
@pukkandan at the time that I made the PR (as you can see in the curl output) the videoPlayerService returned no metadata for these videos. Not sure if this was a temporary glitch that was fixed or some sort of tiered video release system that Paramount Plus/CBS do for some reason, but either way, I think this patch is a good idea in case this happens again for other videos

**Comment by pukkandan** [comment]
> at the time that I made the PR (as you can see in the curl output) the videoPlayerService API endpoint didn't return metadata for these videos.

ah, gotcha. I was confused coz the tests were passing without the PR

File: yt_dlp/extractor/cbs.py [hunk 6]
```diff
@@ -99,25 +156,37 @@ def _extract_video_info(self, content_id, site='cbs', mpx_acc=2198311517):
...
+                query['formats'] = ''  # blank query to check if expired
+                try:
+                    tp_formats, tp_subtitles = self._extract_theplatform_smil(
+                        update_url_query(tp_release_url, query), content_id,
+                        'Downloading %s SMIL data, trying again with another format' % asset_type)
+                except ExtractorError as e:
+                    last_e = e
+                    continue
             formats.extend(tp_formats)
             subtitles = self._merge_subtitles(subtitles, tp_subtitles)
```
Comment (pukkandan) [inline yt_dlp/extractor/cbs.py:170]: Is this correct, or should these formats be ignored?

Comment (llacb47) [reply yt_dlp/extractor/cbs.py:170]: not sure what you're asking here sorry. maybe this answers your question.? https://github.com/yt-dlp/yt-dlp/pull/579#discussion_r680575270

**Pull request merged**